core/test/utils/batch_helpers.hpp

```diff
@@ -106,6 +109,273 @@ std::unique_ptr<MatrixType> generate_random_batch_matrix(
+        if (row < num_rows - 1) {
+            data.nonzeros.emplace_back(row, k, value_type{-1.0});
+            data.nonzeros.emplace_back(row, row + 1, value_type{-1.0});
+        }
+    }
+
+    if (is_hermitian) {
+        gko::utils::make_hpd(data);
+    }
+    data.ensure_row_major_order();
+
+    auto soa_data =
+        gko::device_matrix_data<value_type, index_type>::create_from_host(
+            exec->get_master(), data);
+    auto row_idxs = gko::array<index_type>::const_view(
+                        exec->get_master(), soa_data.get_num_elems(),
+                        soa_data.get_const_row_idxs())
+                        .copy_to_array();
+    auto col_idxs = gko::array<index_type>::const_view(
+                        exec->get_master(), soa_data.get_num_elems(),
+                        soa_data.get_const_col_idxs())
+                        .copy_to_array();
+
+    std::vector<gko::matrix_data<value_type, index_type>> batch_data;
+    batch_data.reserve(num_batch_items);
+    batch_data.emplace_back(data);
+    auto rand_val_dist = std::normal_distribution<>(-0.5, 0.5);
+    for (size_type b = 1; b < num_batch_items; b++) {
+        auto rand_data = fill_random_matrix_data<value_type, index_type>(
+            num_rows, num_cols, row_idxs, col_idxs, rand_val_dist, engine);
+        gko::utils::make_diag_dominant(rand_data);
+        batch_data.emplace_back(rand_data);
+        GKO_ASSERT(rand_data.size == batch_data.at(0).size);
+    }
+    return gko::batch::read<value_type, index_type, MatrixType>(
+        exec, batch_data, std::forward<MatrixArgs>(args)...);
+}
+
+
+template <typename MatrixType>
+struct LinearSystem {
+    using value_type = typename MatrixType::value_type;
+    using multi_vec = batch::MultiVector<value_type>;
+    using real_vec = batch::MultiVector<remove_complex<value_type>>;
+
+    std::shared_ptr<const MatrixType> matrix;
+    std::shared_ptr<multi_vec> rhs;
+    std::shared_ptr<real_vec> rhs_norm;
+    std::shared_ptr<multi_vec> exact_sol;
+};
+
+
+template <typename MatrixType>
+LinearSystem<MatrixType> generate_batch_linear_system(
+    std::shared_ptr<const MatrixType> input_batch_matrix, const int num_rhs)
+{
+    using value_type = typename MatrixType::value_type;
+    using index_type = typename MatrixType::index_type;
+    using multi_vec = batch::MultiVector<value_type>;
+    using real_vec = batch::MultiVector<remove_complex<value_type>>;
+    LinearSystem<MatrixType> sys;
+    sys.matrix = input_batch_matrix;
+    const auto num_batch_items = sys.matrix->get_num_batch_items();
+    const auto num_rows = sys.matrix->get_common_size()[0];
+    auto exec = sys.matrix->get_executor();
+    sys.exact_sol = multi_vec::create(
+        exec, batch_dim<2>(num_batch_items, gko::dim<2>(num_rows, num_rhs)));
+    sys.exact_sol->fill(value_type{2.0});
+
+    sys.rhs = multi_vec::create_with_config_of(sys.exact_sol);
+    // A * x^{*} = b
```

Comment:
```suggestion
    // A * x= b
```

Reply:
`x^{*}` is meant to denote the exact solution, $x^{*}$

Reply:
No, it is general convention to denote the exact solution itself by $x^{*}$ as it is a vector. But I guess it is a bit confusing, so I will clarify that with $x_{exact}$ instead.